x-pack/plugins/fleet/server/routes/agent/actions_handlers.ts

```diff
@@ -39,6 +40,7 @@ export const postNewAgentActionHandlerBuilder = function (
         created_at: new Date().toISOString(),
         ...newAgentAction,
         agents: [agent.id],
```

Comment:
Should we validate the agent is in the current namespace?

Reply:
I believe the call to `actionsService.getAgent` already handles that. I've added an integration test to cover that case.

Reply:
Just checked what is done in `actionsService.getAgent` and it seems there is no namespace validation there.

Reply:
Please correct me if I'm wrong :)

* `actionsService.getAgent` is defined [here](actionsService.getAgent)
* the `postNewAgentActionHandlerBuilder` maps it to `AgentService.getAgentById` [here](https://github.com/elastic/kibana/blob/0298013a784e92ef4efb705f67453a31930c9f42/x-pack/plugins/fleet/server/routes/agent/index.ts#L188)
* and `getAgentById` contains namespace validation [here](https://github.com/elastic/kibana/blob/0298013a784e92ef4efb705f67453a31930c9f42/x-pack/plugins/fleet/server/services/agents/crud.ts#L405-L407)

I've tried it manually as well and I'm seeing 404 for agent ids in other spaces. Am I missing anything?